doc/testing-your-application.rst

```diff
@@ -67,13 +50,60 @@ You can override getters in your test class in the following way::
+``VARNISH_BINARY``      ``getBinary()``           ``varnishd``                                      your Varnish binary
+``VARNISH_PORT``        ``getCachingProxyPort()`` ``6181``                                          port Varnish listens on
+``VARNISH_MGMT_PORT``   ``getVarnishMgmtPort()``  ``6182``                                          Varnish management port
+``VARNISH_CACHE_DIR``   ``getCacheDir()``         ``sys_get_temp_dir()`` + ``foshttpcache-varnish`` directory to use for cache
+``WEB_SERVER_HOSTNAME`` ``getHostName()``                                                           hostname your application can be reached at
+======================= ========================= ================================================= ============================================
 Enable Assertions
-~~~~~~~~~~~~~~~~~
+'''''''''''''''''
 
 For the `assertHit` and `assertMiss` assertions to work, you should add a
 :ref:`custom X-Debug header <varnish_debugging>` to responses served
 by your Varnish.
 
+NginxTestCase
+-------------
+
+Configuration
+'''''''''''''
+
+By default, the ``NginxTestCase`` starts and stops NGINX server for you and deletes
+all cached contents. You have to set your NGINX configuration file. 
+
+These are all the configurations you can change
+
+======================= ========================= =============================================== ==========================================
+Constant                Getter                    Default                        Description
+======================= ========================= =============================================== ==========================================
+``NGINX_FILE``          ``getConfigFile()``                                                       your NGINX configuration file
+``NGINX_BINARY``        ``getBinary()``           ``nginx``                                       your NGINX binary
+``NGINX_PORT``          ``getCachingProxyPort()`` ``8088``                                        port NGINX listens on
+``NGINX_CACHE_PATH``    ``getCacheDir()``         ``sys_get_temp_dir()`` + ``foshttpcache-nginx`` directory to use for cache
+                                                                                                  Must match `proxy_cache_path` directive in 
+                                                                                                  your configuration file.
+``WEB_SERVER_HOSTNAME`` ``getHostName()``                                                         hostname your application can be reached at
+======================= ========================= =============================================== ==========================================
+Enable Assertions
+'''''''''''''''''
+
+For the `assertHit` and `assertMiss` assertions to work, you should add the HTTP 
+header ``X-Cache`` to your responses
```

Comment:
@ddeboer just cleaned this up. its X-Cache-Debug if i am not really mistaken. i will squash your commits and adjust this and merge.